### PR TITLE
implemented readability index for german texts

### PR DESCRIPTION
**Pull request by RolandColored**
I added a new formula to the class _TextStats_ to open up this functionality to other languages.
So far I have not touched the method _readability_stats_, because I'm unsure how you would like to handle multiple languages. Still, I'm open to extend it after further discussions like:

- Would you like to return different structured dictionaries based on language?
- Would you like to split the method into language-dependant (private sub-)methods?

Maybe you already had some design ideas in mind you could share with me.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation, and I have updated it accordingly.

### CODE REVIEW

File: textacy/text_stats.py [hunk 1]
```diff
@@ -85,3 +85,20 @@ def coleman_liau_index(n_chars, n_words, n_sents):
...
+
+
+def wiener_sachtextformel(n_words, n_sents, n_polysyllable_words, n_singlesyllable_words, n_long_words, variant=1):
+    """https://de.wikipedia.org/wiki/Lesbarkeitsindex#Wiener_Sachtextformel"""
+    ms = n_polysyllable_words / n_words * 100
+    sl = n_words / n_sents
+    iw = n_long_words / n_words * 100
+    es = n_singlesyllable_words / n_words * 100
+    if variant == 1:
+        return 0.195 * ms + 0.1672 * sl + 0.1297 * iw - 0.0327 * es - 0.875
```
Comment (bdewilde) [inline textacy/text_stats.py]: minor typo: 0.195 should be 0.1935

File: textacy/text_stats.py [hunk 2]
```diff
@@ -85,3 +85,20 @@ def coleman_liau_index(n_chars, n_words, n_sents):
...
+    elif variant == 2:
+        return 0.2007 * ms + 0.1682 * sl + 0.1473 * iw - 2.779
```
Comment (bdewilde) [inline textacy/text_stats.py]: another minor typo: 0.1473 should be 0.1373

File: textacy/text_stats.py [hunk 3]
```diff
@@ -85,3 +85,20 @@ def coleman_liau_index(n_chars, n_words, n_sents):
 def automated_readability_index(n_chars, n_words, n_sents):
     """https://en.wikipedia.org/wiki/Automated_readability_index"""
     return 4.71 * (n_chars / n_words) + 0.5 * (n_words / n_sents) - 21.43
+
+
+def wiener_sachtextformel(n_words, n_sents, n_polysyllable_words, n_singlesyllable_words, n_long_words, variant=1):
```
Comment (bdewilde) [inline textacy/text_stats.py]: Would you mind adding parentheses around all of the subexpressions to make the order of operations more obvious? I don't see any errors, but I always have to remind myself how expressions like `X / Y * Z` actually evaluate.

**Comment by RolandColored** [comment]
The changes got pushed.

For handling multiple languages I would go for a simple solution in the beginning: Extending the dictionary of statistics with language dependent readability stats. So existing code will not break.

A more complex approach could make use of delegation or even inheritance. All readability methods go into new classes EnTextStats or DeTextStats while TextStats still calculates the common text statistics.

**Comment by bdewilde** [comment]
Looks good! Thanks a ton for submitting. I'll merge these changes in. 🙌 

Separately, I'm going to work out a nice way of handling different languages. I'll want to think about it holistically, and come up with a solution that's consistent across the whole package and not just within a module. Could be a big project...

**Pull request merged**